src/udev/udev-node.c

```diff
@@ -188,28 +201,116 @@ static int link_find_prioritized(sd_device *dev, bool add, const char *stackdir,
         return 0;
 }
 
+static int init_link_semaphores(const char *path) {
+        key_t key = ftok(path, 0);
+        int semid;
+
+        semid = semget(key, N_SEMAPHORES, 0600|IPC_CREAT|IPC_EXCL);
+        if (semid != -1) {
+                unsigned short val[N_SEMAPHORES];
+                int i;
+                struct sembuf dummy_op[]  = {
+                        { .sem_op = -1, },
+                        { .sem_op = 1, },
+                };
+
+                for (i = 0; i < N_SEMAPHORES; i++)
+                        val[i] = 1;
+                if (semctl(semid, 0, SETALL, val) == -1)
+                        return log_error_errno(-errno, "Failed to initialize semaphores: %m");
+
+                /* Dummy semop to set sem_otime */
+                if (semop(semid, dummy_op, (sizeof(dummy_op)/sizeof(*dummy_op))) == -1)
+                        return log_error_errno(-errno, "Failed to set sem_otime: %m");
+                return semid;
+        } else {
+                const unsigned int RETRIES = 10, SLEEP_US = 10000;
```

Comment:
we generally call "unsigned int" just "unsigned".

Also, so far we have avoid using "const" on anything that is not a pointer, since it gives useless guarantees here... maybe make this a macro?